hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterServiceImpl.java

```diff
@@ -345,16 +352,28 @@ public void merge(Address newTargetAddress) {
     public void reset() {
         lock.lock();
         try {
-            membershipManager.reset();
-            clusterHeartbeatManager.reset();
-            clusterStateManager.reset();
-            clusterJoinManager.reset();
+            resetLocalMemberUuid();
             resetClusterId();
+            clearInternalState();
         } finally {
             lock.unlock();
         }
     }
 
+    private void resetLocalMemberUuid() {
+        assert lock.isHeldByCurrentThread() : "Called without holding cluster service lock!";
+        assert !isJoined() : "Cannot reset local member uuid when joined.";
```

Comment:
`resetLocalMemberUuid()` is already a private method and it's called in a controlled flow. Assertion is mostly to document the behaviour, not for correctness of the method.